packages/ibm-products/src/components/UserAvatar/UserAvatar.js

```diff
@@ -46,43 +46,51 @@ const componentName = 'UserAvatar';
  * TODO: A description of the component.
  */
 
-const defaults = {
-  renderIcon: () => <User size={32} />,
-  tooltipAlignment: 'bottom',
-  tooltipText: 'Thomas J. Watson',
-};
-
 export let UserAvatar = React.forwardRef(
   (
     {
       // The component props, in alphabetical order (for consistency).
       backgroundColor,
       className,
+      name,
       /* TODO: add other props for UserAvatar, with default values if needed */
-      renderIcon = defaults.renderIcon,
-      tooltipText = defaults.tooltipText,
-      tooltipAlignment = defaults.tooltipAlignment,
+      renderIcon,
+      size,
+      tooltipText,
+      tooltipAlignment,
       // Collect any other property values passed in.
       ...rest
     },
     ref
   ) => {
     const carbonPrefix = usePrefix();
-    const icons = {
-      user: {
-        md: <User size={32} />,
-      },
-      group: {
-        md: <Group size={32} />,
-      },
+    const iconSize = {
+      sm: 16,
+      md: 20,
+      lg: 24,
+      xl: 32,
+    };
+    const formatInitials = () => {
+      if (name.length === 2) {
+        return name;
+      }
+      // RegEx takes in the display name and returns the first and last name. Thomas Watson and Thomas J. Watson
+      // both return JW.
+      return name
+        .match(/(^\S\S?|\b\S)?/g)
+        .join('')
+        .match(/(^\S|\S$)?/g)
+        .join('')
+        .toUpperCase();
     };
     const getItem = (renderIcon) => {
-      if (renderIcon === User) {
-        return icons.user['md'];
-      } else if (renderIcon === Group) {
-        return icons.group['md'];
+      let Iconcomponent = renderIcon;
```

Comment:
this line isn't necessary when we declare `renderIcon` as a component in the props. and since, as i mentioned before, the user isn't passing in any props directly to icon we don't need to return a function. we can just do something like this
```
renderIcon: RenderIcon,
...
const getContent = () => {
  const iconProps = {
    size: iconSize[size],
  };
  if (RenderIcon) {
    return <RenderIcon {...iconProps} />;
  }
  if (name) {
    return formatInitials;
  }
  return <User {...iconProps} />;
};
```

Reply:
makes things even easier!